src/main/scala/mesosphere/marathon/upgrade/DeploymentPlan.scala

```diff
@@ -305,7 +307,18 @@ object DeploymentPlan {
     result
   }
 
-  implicit lazy val deploymentPlanIsValid: Validator[DeploymentPlan] = validator[DeploymentPlan] { plan =>
-    plan.createdOrUpdatedApps as "app" is every(valid(AppDefinition.updateIsValid(plan.original)))
+  def deploymentPlanValidator(conf: MarathonConf): Validator[DeploymentPlan] = {
+    val maxSize = conf.zooKeeperMaxNodeSize()
+    val notBeTooBig = isTrue[DeploymentPlan](s"The deployment plan size can't be larger than $maxSize bytes.") { plan =>
```

Comment:
Perhaps something like:

>
The way we persist data in ZooKeeper would lead to a ZK node size greater than $maxSize.
You can adjust this value via `--zk_max_node_size`, but make sure this value is supported! 
See: http://zookeeper.apache.org/doc/r3.3.1/zookeeperAdmin.html#sc_configuration